chore: fix all Typescript warnings

gitstart-calcom opened the pull request:
Fixes #8351 

yarn lint: 
![image](https://user-images.githubusercontent.com/87545086/233743125-89170472-14ad-480c-b8ab-6c8f8bcd262a.png)

emrysal commented on apps/web/components/AppListCard.tsx:
```diff
@@ -60,7 +60,7 @@ export default function AppListCard(props: AppListCardProps) {
       }
     };
-  }, []);
+  }, [highlight, router, shouldHighlight]);
```
This is very risky, we can't merge this - I pushed a suggestion to https://github.com/calcom/cal.com/pull/8621 which @keithwillcode will evaluate, after which we can re-run the fixer and hopefully get the desired fixups without breaking anything.

keithwillcode replied:
@emrysal My thoughts [here](https://github.com/calcom/cal.com/pull/8621/files#r1192590344).

zomars commented on apps/web/components/apps/layouts/InstalledAppsLayout.tsx:
```diff
@@ -2,60 +2,12 @@ import type { ComponentProps } from "react";
-import type { HorizontalTabItemProps, VerticalTabItemProps } from "@calcom/ui";
-import { BarChart, Calendar, CreditCard, Grid, Share2, Video } from "@calcom/ui/components/icon";
-
-const tabs: (VerticalTabItemProps | HorizontalTabItemProps)[] = [
```
Is this intended?

gitstart-calcom replied:
yes, we removed these because it was unused code

gitstart-calcom replied:
should we comment them instead?

zomars replied:
It's ok if it's really unused

zomars commented on apps/web/pages/auth/saml-idp.tsx:
```diff
@@ -19,7 +19,7 @@ export default function Page() {
       code,
     });
-  }, []);
+  }, [router.isReady, router.query]);
```
Router.query is an object. I think this will cause unintended behavior

zomars commented on apps/web/pages/auth/sso/direct.tsx:
```diff
@@ -29,7 +29,7 @@ export default function Page({ samlTenantID, samlProductID }: inferSSRProps<type
       { tenant: samlTenantID, product: samlProductID }
     );
-  }, []);
+  }, [samlProductID, samlTenantID]);
```
This changes the useEffect behavior. Let's be careful with these changes.

zomars commented on apps/web/pages/auth/sso/direct.tsx:
```diff
@@ -18,7 +18,7 @@ export default function Page({ samlTenantID, samlProductID }: inferSSRProps<type
       router.push("/auth/login");
     }
-  }, []);
+  }, [router]);
```
This changes the useEffect behavior. Let's be careful with these changes.

zomars reviewed (dismissed):
Agreed with @emrysal we cannot trust auto fixes here. Most useEffect calls with empty dependencies are intentional. Let's triple check those please.

gitstart committed:
[CAL-1517] fix all Typescript warnings

PeerRich committed:
Merge branch 'main' into CALCOM-8351

PeerRich committed:
Merge branch 'main' into CALCOM-8351

gitstart committed:
Merge commit '1b68cc139f212ab2d3fc2490844561b1a587ad5b' into CALCOM-8351

gitstart committed:
solve conflicts

gitstart-calcom committed:
Merge branch 'main' into CALCOM-8351

emrysal committed:
Merged with main

emrysal committed:
Merge branch 'main' into CALCOM-8351

emrysal committed:
Update stripeCheckoutSession.handler.ts

Parse is a guard, so even though the variable is unused the parse itself is needed.

emrysal committed:
Update ToolbarPlugin.tsx

Don't change dependency tree

emrysal reviewed (approved):
Applied a couple of pending fixes, if tests pass it's on auto-merge, nice work.

emrysal committed:
Better fix to newMetadata warn
